code/modules/1713/fiatcurrency.dm

```diff
@@ -0,0 +1,178 @@
+		message_admins("Fiat currency was created without a fiat_id var!")
+
+/obj/item/stack/money/fiat/attackby(obj/item/stack/money/fiat/W as obj, mob/user as mob)
+	if (istype(W, type) && W.fiat_id == src.fiat_id)
+		var/obj/item/stack/S = W
+		merge(S)
+		S.update_icon()
+		S.update_strings()
+		src.update_icon()
+		src.update_strings()
+		spawn(0) //give the stacks a chance to delete themselves if necessary
+		if (S && usr.using_object == S)
+			S.interact(usr)
+		if (src && usr.using_object == src)
+			interact(usr)
+	else
+		return ..()
+
+/obj/item/stack/money/fiat/split(var/_amount, fiat_id)
+	if (!amount)
+		return null
+	if (_amount)
+		var/obj/item/stack/money/fiat/S = new(src.loc, _amount, fiat_id)
+		return S
+	return FALSE
+
+/obj/structure/money_printer
+	name = "money printer"
+	desc = "Used to print money. It currently contains 0 cloth."
+	icon = 'icons/obj/structures.dmi'
+	icon_state = "printingpress0"
+	anchored = TRUE
+	density = TRUE
+	not_movable = FALSE
+	not_disassemblable = TRUE
+	var/fiat_id
+	var/cloth
+
+/obj/structure/money_printer/proc/print_money(user)
+	var/nm
+	var/num = input(user, "How much money do you want to print?","Printing", nm) as num
+	if (num == 0)
+		return
+	else
+		for(var/i in fiat.currency_list)
+			var/cloth_remaining = (cloth - num*0.01)
+			if (fiat_id && cloth_remaining >= 0)
+				cloth = cloth_remaining
+				desc = "Used to print money. It currently contains [cloth] cloth."
+				new/obj/item/stack/money/fiat(loc, num, fiat_id)
+				playsound(loc, 'sound/machines/vending_drop.ogg', 100, TRUE)
+				return
+			else if (cloth_remaining < 0)
+				to_chat(user, "You do not have enough cloth in the machine to produce that much currency! The maximum you can currently produce is [cloth*100]")
+				return
+			else if (!fiat_id)
+				to_chat(user, "That currency does not exist.")
+				return
+
+/obj/structure/money_printer/proc/display_ui(user)
+	var/choice = WWinput(user, "Pick an option", "Money Printer", "Cancel", list("Cancel", "Print money"))
+	if (choice == "Cancel")
+		return
+	else if (choice == "Print money")
+		print_money(user)
+		return
+
+/obj/structure/money_printer/attack_hand(var/mob/living/human/user as mob)
+	if(fiat_id)
+		display_ui(user)
+	else
+		var/cd
+		var/currency_code = input(user, "Input the ID for your currency. You can use any text.","Currency Creation", cd) as text
+		var/currency_exists = 0
+		var/currency_revival = 0
+		for(var/codes in fiat.currency_list)
+			if(codes == currency_code)
+				if (fiat.currency_list[codes][5] == 1)
+					to_chat(user, "That currency already exists.")
+					currency_exists = 1
+					break
+				else
+					currency_revival = 1
+					break
+
+		if(currency_exists == 0)
+			fiat_id = currency_code
+			if (currency_revival == 1)
+				fiat.currency_list[fiat_id][5] = 1
+				name = "[fiat.currency_list[fiat_id][1]] printer"
+				to_chat(user, "You rebuild the printer for the [fiat.currency_list[fiat_id][1]].")
```

Comment:
could use a tochat... SPAN_NOTICE("You rebuild the printer for the [fiat.currency_list[fiat_id][1]].")